server/db/create-shapes.ts

```diff
@@ -81,50 +94,55 @@ class CreateShapes {
     })
   }
 
-  upload(container, directory) {
+  upload(container: string, directory: string) {
     return new Promise((resolve, reject) => {
-      if (global.config.shapesSkip === true) {
+      if (config.shapesSkip === true) {
         log('Skipping Shapes Upload.')
         return resolve()
       }
       let total = 0
-      const uploadSingle = (files, index, callback) => {
+      const uploadSingle = (files: string[], index: number, callback: any) => {
         if (index === files.length) {
           log(`${container.magenta}:`, 'Upload Complete!', total, 'Shapes.')
           return resolve()
         }
 
         const fileName = files[index]
-        const key = `${global.config.prefix}/${directory
+        const key = `${config.prefix}/${directory
           .split('/')
           .slice(-1)[0]
           .replace('_', '-')
           .replace('.', '-')}/${fileName}`
-        const fileLocation = path.resolve(directory, fileName)
-        this.storageSvc.uploadFile(container, key, fileLocation, error => {
-          if (error) {
-            console.error(
-              `${container.magenta}:`,
-              'Could not upload shape.',
-              error
-            )
-          }
-          total += 1
-          if (total % 100 === 0) {
-            log(`${container.magenta}:`, 'Uploaded', total, 'Shapes.')
+        const fileLocation = _resolve(directory, fileName)
+        this.storageSvc.uploadFile(
+          container,
+          key,
+          fileLocation,
+          (error: any) => {
+            if (error) {
+              console.error(
+                `${container.magenta}:`,
+                'Could not upload shape.',
+                error
+              )
+            }
+            total += 1
+            if (total % 100 === 0) {
+              log(`${container.magenta}:`, 'Uploaded', total, 'Shapes.')
+            }
+            callback(files, index + 1, callback)
```

Comment:
this is my fault, and it's nasty :( we should fix this at some point rather than recursive hell